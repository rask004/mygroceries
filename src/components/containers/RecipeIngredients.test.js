import React from 'react';
import {create} from 'react-test-renderer';

import IngredientsList from './RecipeIngredients';


describe('RecipeIngredients Component render tests', () => {

    const ingredients = [
        {
            id: 1,
            name: "apple",
            coo: "new zealand",
            quantity: 2,
            unit: "each",
        },
        {
            id: 2,
            name: "orange",
            coo: "australia",
            quantity: 3,
            unit: "each",
        },
    ];
    
    const component = create(<IngredientsList
            ingredients={ingredients}
            addIngredient={jest.fn()}
            removeIngredient={jest.fn()}
    />);

    test('renders correctly', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('renders the given ingredients', () => {
        const rootInstance = component.root;
        const listitems = rootInstance.findAllByType("li");
        for (let i = 0; i < ingredients.length; i++) {
            const item = listitems[i];
            const contents = item.findAllByProps({className: "detail-content"});
            // actually 2 child rendered components per each declared component.
            // declared component, and clone with 'classes'
            // we only need the declared component
            expect(contents[0].props.children).toBe(ingredients[i].name);
            expect(contents[2].props.children).toBe(ingredients[i].coo);
            expect(contents[4].props.children).toBe(ingredients[i].quantity);
            expect(contents[6].props.children).toBe(ingredients[i].unit);
        }
    });
});


describe('RecipeIngredients Component behaviour tests', () => {

    const ingredients = [
        {
            id: 1,
            name: "apple",
            coo: "new zealand",
            quantity: 2,
            unit: "each",
        },
        {
            id: 2,
            name: "orange",
            coo: "australia",
            quantity: 3,
            unit: "each",
        },
    ];

    const component = create(<IngredientsList
        ingredients={ingredients}
        addIngredient={jest.fn()}
        removeIngredient={jest.fn()}
    />);

    test('add button click calls addIngredient', () => {
        const rootInstance = component.root;

        const addButton = rootInstance.findByProps({className: "action-add"});
        addButton.props.onClick();
        expect(rootInstance.props.addIngredient).toBeCalled();
    });

    test('remove button click calls removeIngredient', () => {
        const idIngredientRemove = ingredients[0].id;

        const rootInstance = component.root;
        const listitem = rootInstance.findAllByType("li")[0];
        const removeButton = listitem.findByProps({className: "action-remove"});
        removeButton.props.onClick();

        expect(rootInstance.props.removeIngredient).toBeCalledWith(idIngredientRemove);
    });
});
