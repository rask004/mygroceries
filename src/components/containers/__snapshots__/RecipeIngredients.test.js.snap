// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeIngredients Component render tests renders correctly 1`] = `
<div
  className="detail-ingredients"
>
  <h2
    className="MuiTypography-root-1 MuiTypography-title-7 MuiTypography-gutterBottom-28"
  >
    Ingredients
  </h2>
  <ul
    className="MuiList-root-37 MuiList-padding-38"
  >
    <li>
      <span
        className="MuiTypography-root-1 MuiTypography-body1-10"
      >
        <div
          className="MuiGrid-container-41"
        >
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-4-101 MuiGrid-grid-lg-5-116 detail-content"
          >
            apple
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-3-100 MuiGrid-grid-lg-4-115 detail-content"
          >
            new zealand
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-2-99 MuiGrid-grid-lg-1-112 detail-content"
          >
            2
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-2-99 MuiGrid-grid-lg-1-112 detail-content"
          >
            each
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-1-98 MuiGrid-grid-lg-1-112"
          >
            <button
              className="MuiButtonBase-root-164 MuiButton-root-138 MuiButton-contained-149 MuiButton-containedPrimary-150 MuiButton-raised-152 MuiButton-raisedPrimary-153 action-remove"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-139"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-167"
              />
            </button>
          </span>
        </div>
      </span>
    </li>
    <li>
      <span
        className="MuiTypography-root-1 MuiTypography-body1-10"
      >
        <div
          className="MuiGrid-container-41"
        >
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-4-101 MuiGrid-grid-lg-5-116 detail-content"
          >
            orange
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-3-100 MuiGrid-grid-lg-4-115 detail-content"
          >
            australia
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-2-99 MuiGrid-grid-lg-1-112 detail-content"
          >
            3
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-2-99 MuiGrid-grid-lg-1-112 detail-content"
          >
            each
          </span>
          <span
            className="MuiGrid-item-42 MuiGrid-grid-xs-12-81 MuiGrid-grid-sm-6-89 MuiGrid-grid-md-1-98 MuiGrid-grid-lg-1-112"
          >
            <button
              className="MuiButtonBase-root-164 MuiButton-root-138 MuiButton-contained-149 MuiButton-containedPrimary-150 MuiButton-raised-152 MuiButton-raisedPrimary-153 action-remove"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-139"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-167"
              />
            </button>
          </span>
        </div>
      </span>
    </li>
  </ul>
  <button
    className="MuiButtonBase-root-164 MuiButton-root-138 MuiButton-contained-149 MuiButton-containedPrimary-150 MuiButton-raised-152 MuiButton-raisedPrimary-153 action-add"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-139"
    >
      Add Ingredient...
    </span>
    <span
      className="MuiTouchRipple-root-167"
    />
  </button>
</div>
`;
